tests/rptest/services/compatibility/compat_example.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import sys
+import time
+from ducktape.services.background_thread import BackgroundThreadService
+
+
+class CompatExample(BackgroundThreadService):
+    def __init__(self, context):
+        super(CompatExample, self).__init__(context, num_nodes=1)
+        self._cmds = None
+        self._condition = None
+        self._condition_met = False
+        self._timeout = 10
+        self._node = None
+
+    def _worker(self, idx, node):
+        start_time = time.time()
+        if self._node is None:
+            self._node = node
+
+        #Run the example until the condition is met
+        output_iter = self._node.account.ssh_capture(self._cmds[0])
+        while not self._condition_met and time.time() < start_time + self._timeout:
+            line = next(output_iter)
+            self._condition_met = self._condition(line.rstrip())
+
+    #ok - used to determine if the condition is met
+    def ok(self):
+        return self._condition_met
+
+    #set_example - assigns the command, stop condition, and node
+    #for an example
+    def set_example(self, cmds, condition, node=None):
+        self._cmds = cmds
+        self._condition = condition
+        self._condition_met = False
+        self._node = node
```

Comment:
allowing the node to be set here feels like a leaky abstraction. how can we avoid that? can these parameters be set in the constructor and remain unchanged?